resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -328,4 +328,19 @@ package object config {
     CACHED_FILES_TYPES,
     CACHED_CONF_ARCHIVE)
 
+  /* YARN allocator-level blacklisting related config entries. */
+  private[spark] val YARN_EXECUTOR_LAUNCH_BLACKLIST_ENABLED =
+    ConfigBuilder("spark.yarn.blacklist.executor.launch.blacklisting.enabled")
+      .booleanConf
+      .createOptional
```

Comment:
why createOptional vs createwithdefault?  Depending on what we decide for limit, worse case is we just have default false.

Reply:
I followed the existing pattern used for BLACKLIST_ENABLED:
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/config/package.scala#L164

But you are right createWithDefault is more adequate here. I will change this.